Amplitude/Amplitude.m

```diff
@@ -1313,12 +1331,22 @@ - (void)setEventUploadMaxBatchSize:(int) eventUploadMaxBatchSize
 
 - (BOOL)optOut
 {
+    if (_apiKey == nil) {
+        AMPLITUDE_ERROR(@"ERROR: apiKey cannot be nil or empty, set apiKey with initializeApiKey: before toggling and fetching optOut");
+        return NO;
+    }
     return [[self.dbHelper getLongValue:OPT_OUT] boolValue];
 }
 
 - (void)setDeviceId:(NSString*)deviceId
 {
+    if (_apiKey == nil) {
```

Comment:
As per discussion, if we modify the `instance` call, then customer would potentially have to update every single line where they call Amplitude, so we're going with `[[[Amplitude instance] withApiKey:apiKey] initialize];` instead